test/SerialIOTest.cpp

```diff
@@ -1445,6 +1445,40 @@ TEST_CASE( "hdf5_110_optional_paths", "[serial][hdf5]" )
         REQUIRE(s.iterations[1].particles.empty());
     }
 }
+
+TEST_CASE( "hdf5_constant_scalar", "[serial][hdf5]" )
```

Comment:
@C0nsultant should we also add a test for a non-scalar record that only has a constant record component (but not all record components are constant)? Or is this case already covered somewhere else?

Reply:
Are you talking about a `Record`with mixed constant and non-constant `RecordComponent`s? I don't think we have such a case yet.

Reply:
Yep, that's what I refer to.

Reply:
I included a little something for this.